dask/dataframe/io/tests/test_sql.py

```diff
@@ -332,3 +335,93 @@ def test_no_character_index_without_divisions(db):
     # attempt to read the sql table with a character index and no divisions
     with pytest.raises(TypeError):
         read_sql_table("test", db, npartitions=2, index_col="name", divisions=None)
+
+
+@contextmanager
+def tmp_db_uri():
+    with tmpfile() as f:
+        yield "sqlite:///%s" % f
```

Comment:
Do you need to use a file on disk, or can you use `:memory:`?

Reply:
Apparently, memory isn't happy with access from multiple threads